### PR TITLE
Change the application access authentication flow

### PR DESCRIPTION
**Pull request by ryanclark**
This is the frontend changes for this PR - https://github.com/gravitational/teleport/pull/17592

This sets a custom `X-Cookie-Value` header on the request to `/x-teleport-auth` instead of putting it in the `POST` body.

We also remove a lot of the redirects (and the code that goes with it) as it's no longer required with the new authentication flow in the Teleport PR.

### CODE REVIEW

**Comment by ikkisoft** [comment]
LGMT. Comments left in [#17592](https://github.com/gravitational/teleport/pull/17592)

File: packages/teleport/src/AppLauncher/AppLauncher.tsx [hunk 1]
```diff
@@ -14,25 +14,77 @@ See the License for the specific language governing permissions and
...
+        method: 'POST',
+        credentials: 'include',
+        headers: {
+          'X-Cookie-Value': session.cookieValue,
+          'X-Subject-Cookie-Value': session.subjectCookieValue,
+        },
+      });
+
+      window.location.replace(`https://${fqdn}${port}`);
+    } catch (err) {
```
Comment (kimlisa) [inline packages/teleport/src/AppLauncher/AppLauncher.tsx:57]:
```suggestion
    } catch ((err: Error)) {
```

**Review by kimlisa — APPROVED** [review]
> We also remove a lot of the redirects (and the code that goes with it) as it's no longer required with the new authentication flow in the Teleport PR.

that means like url params like `awsrole` and `state` right? can we also delete `useAppLaunder` file since it doesn't look like it's being used. also does the backend PR delete the code that does the redirecting? eg `redirectToLauncher`, before I saw the above comment, i thought we missed these params

File: packages/teleport/src/AppLauncher/AppLauncher.tsx (hunk 1 above)
Comment (ryanclark) [reply packages/teleport/src/AppLauncher/AppLauncher.tsx:57]: It's never guaranteed what is being thrown is of type `Error` -
image

Comment (kimlisa) [reply packages/teleport/src/AppLauncher/AppLauncher.tsx:57]: ah ok, then should we do this (create a helper func perhaps): https://github.com/gravitational/webapps/blob/5458f66bccbf7e3df879e98e7ee8096bb0f7665b/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.ts#L188

**Comment by ryanclark** [comment]
> that means like url params like awsrole and state right? can we also delete useAppLaunder file since it doesn't look like it's being used. also does the backend PR delete the code that does the redirecting? eg redirectToLauncher, before I saw the above comment, i thought we missed these params

`state` is no longer needed - the backend changes remove the `?state=` dance

`arn` is preserved and passed through to `createAppSession` if it's present

image

I forgot about `?path=` and have updated the PR to redirect to the given path if it's in the query params - nice catch!

File: packages/teleport/src/AppLauncher/AppLauncher.tsx (hunk 1 above)
Comment (ryanclark) [reply packages/teleport/src/AppLauncher/AppLauncher.tsx:69]: I don't love that approach as it's assuming if we don't get an `Error`, that the thing thrown can be converted to a string. If it's an object that's thrown, that would put `[object Object]` into the message.

I've updated this catch to check for an instance of an error and use the error message provided if so. If not, it'll just fallback to a generic "Something went wrong"

**Review by hatched — APPROVED** [review]

**Pull request merged**